src/format/scan.rs

```diff
@@ -14,7 +14,7 @@ use crate::Weekday;
 /// More than `max` digits are consumed up to the first `max` digits.
 /// Any number that does not fit in `i64` is an error.
 #[inline]
-pub(super) fn number(s: &str, min: usize, max: usize) -> ParseResult<(&str, i64)> {
+pub(super) fn number(s: &str, min: usize, max: usize, positive: bool) -> ParseResult<(&str, i64)> {
```

Comment:
What are we trying to do here? Why do we need to support negative timestamps? Having to pass in `true` for `positive` all over the place doesn't look great.

Reply:
> Why do we need to support negative timestamps?

We support negative timestamps everywhere, it seems like an oversight to not support them with the `%s` formatting specifier when parsing.

> Having to pass in `true` for `positive` all over the place doesn't look great.

My problem was that I wanted to do a complete fix instead of partial. I.e. to also support parsing `i64::MIN` (which can be stored in `Parsed` even though it is out of range for `NaiveDateTime`, and is something I expect users may test against). Because `-i64::MIN` can't be represented in an `i64` this function should either return an `u64`, or it must return a value with the correct sign. Returning with the correct sign was the minimal change.

Reply:
:+1: I'll do that.